### ISSUE TITLE
Nothing happens when pointing to CP2077 exe file with wait_on_process and toggle_hdr = 1

### ISSUE THREAD

**Issue by Rabusek** [lyckantropen/moonlight_hdr_launcher#1]
Here's my full config file:

```
[options]
# command to launch when the launcher is executed  
launcher_exe = 'C:\Program Files (x86)\GOG Galaxy\Games\Cyberpunk 2077\bin\x64\Cyberpunk2077.exe'
wait_on_process = 1
toggle_hdr = 1
```

Here's all the log that is given:
```
2020-12-22 15:31:23: argv[0]: C:\Program Files\moonlight_hdr_launcher\MassEffectAndromeda.exe
2020-12-22 15:31:23: Found config file: C:\Program Files\moonlight_hdr_launcher\moonlight_hdr_launcher.ini
```

I tried:
- putting no quotes in the exe launcher path
- putting double quotes
- putting single quotes (pasted into the issue above)
- trying with gaemstream_launchapd worked, but did not turn on HDR for me (i have HDR-less monitor)

Any tips to get through these?

**Comment by lyckantropen** [comment]
Quotes don't matter, better leave it without any.

I don't own Cyberpunk 2077 so I can't test this. If you open a command line prompt and just launch `C:\Program Files (x86)\GOG Galaxy\Games\Cyberpunk 2077\bin\x64\Cyberpunk2077.exe`, does it run the game **and** does it not go back to the prompt immediately?

If it goes back to prompt, the `wait_on_process` option won't do anything.

Running with or without gamestream_launchpad should have the same effect, so if it doesn't work it under gamestream_launchpad, it probably won't.

I will take a look at tmac666's solution for non-HDR monitor owners and see if I can deduce what other tricks I should do.

**Comment by lyckantropen** [comment]
On second thought there should be more messages in the log file. Please download the newest release and test again. There are no new features, just more error messages.

**Comment by lyckantropen** [comment]
On non-HDR monitor owners, as per [tmac666's instructions](https://www.nvidia.com/en-us/geforce/forums/user/213217/338010/hdr-fix-for-gamestream-and-steam-big-picture-mode/), it appears that the only way to support HDR is to launch Steam Big Picture through the `MassEffectAndromeda.exe` launcher. The same caveat applies: if the game requires OS-enabled HDR, it won't work. I have been doing it and it works for some games.

Here's `gamestream_steam_bp.ini`:

```ini
[LAUNCHER]
launcher_path = %%programfiles(x86)%%\Steam\steam.exe steam://open/bigpicture Fullscreen
# set it to something other than Steam because gamestream_launchpad can detect multiple windows and will close prematurely
launcher_window_name = NotSteam

[BACKGROUND]
# List as many exe's or bat's as you want here. They will run at the start of the GameStream session and be killed at the end.
# background_exe_1 = C:\Program Files (x86)\JoyToKey\JoyToKey.exe
# background_exe_2 = C:\WINDOWS\system32\mspaint.exe
# background_exe_1 = C:\Program Files\moonlight_hdr_launcher\hdr_toggle.exe

[SETTINGS]
debug = 0
sleep_on_exit = 0
close_watch_method = window
```

Here's `moonlight_hdr_launcher.ini`:

```ini
[options]
# launch Steam Big Picture using gamestream_launchpad
# please adjust resolution
launcher_exe = gamestream_launchpad.exe 2560 1440 gamestream_steam_bp.ini
wait_on_process = 1
toggle_hdr = 1
```

Please bear in mind that you will have to force-quit the launcher in Shield/Moonlight as `gamestream_launchpad` doesn't handle quitting Steam Big Picture well.

**Comment by Rabusek** [comment]
So to start off, when i start installing new version, even when I clean up previous installs, including GF Now, I am constantly getting an error that the Andromeda game has not been found. I assume this is something i can skip so i just moved forward.

Tried opening the game via GF Experience / Shield / Moonlight - nothing, no logs. 
Opening it up manually as admin - produces two lines of logs:
```
2020-12-26 01:38:08: argv[0]: C:\Program Files\moonlight_hdr_launcher\MassEffectAndromeda.exe
2020-12-26 01:38:08: Found config file: C:\Program Files\moonlight_hdr_launcher\moonlight_hdr_launcher.ini
```

I assume this has something to do with restricted Program Files folder access - thus, i tried moving it outside the folder. There's a slight improvement (log file is being written to now even when launched from inside GF Experience), but nothing happens still:

```
2020-12-26 01:42:36: argv[0]: MassEffectAndromeda.exe
2020-12-26 01:42:36: Found config file: moonlight_hdr_launcher.ini
```

**Comment by lyckantropen** [comment]
Yeah OS level HDR probably won't work. I hear there are HDMI dongles out there that have HDR support, maybe that will work.

Please reinstall using latest installer by right-clicking and running it as Administrator, then see if running the MassEffectAndromeda.exe file directly does what's expected.

I'll go back to fixing it when I come back from Christmas. I will add installer logging. I'll be grateful if you stick around for trying out things so I can provide a solution.

**Comment by Rabusek** [comment]
> Yeah OS level HDR probably won't work. I hear there are HDMI dongles out there that have HDR support, maybe that will work.

I've ordered one and will get it by Wednesday (took express Amazon delivery), so hopefully by then will be able to check.

> Please reinstall using latest installer by right-clicking and running it as Administrator, then see if running the MassEffectAndromeda.exe file directly does what's expected.

I tried that as well with nothing happening (unless i left it configured with GoG, then it at least opened up GoG).

**Comment by lyckantropen** [comment]
Hi, please install the latest release (1.0.3) and if problems persist, attach the `moonlight_hdr_launcher_install.log` file that will be created alongside `install.exe` as well as `moonlight_hdr_launcher_log.txt` from `C:\Program Files\moonlight_hdr_launcher\`.

Bear in mind that installation will overwrite your `moonlight_hdr_launcher.ini` file, so make a backup.

**Comment by lyckantropen** [comment]
@Rabusek Most of the issues also highlighted in #2 have been resolved in release 1.0.8. Let me know when you get to test it again.

**Comment by lyckantropen** [comment]
The conclusion is that Cyberpunk 2077 is an OS-level HDR game and therefore requires either an HDR-enabled monitor or an HDR-enabled HDMI or DP dongle. An example product is the [FUERAN DP - DisplayPort Display Emulator EDID Emulator Plug](https://www.amazon.com/FUERAN-DP-DisplayPort-emulator-2560x1600/dp/B071CGCTMY) (no endorsement).

Otherwise I'm closing this due to inactivity.
